firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -290,30 +258,106 @@ public void testGetId_PersistedFidError_BackendOk() throws InterruptedException
     }
   }
 
+  @Test
+  public void testGetId_fidRegistrationUncheckedException_statusUpdated() throws Exception {
+    // Mocking unchecked exception on FIS createFirebaseInstallation
+    when(mockClient.createFirebaseInstallation(anyString(), anyString(), anyString(), anyString()))
+        .thenAnswer(
+            invocation -> {
+              throw new InterruptedException();
+            });
+
+    FirebaseInstallations firebaseInstallations =
+        new FirebaseInstallations(executor, firebaseApp, mockClient, persistedFid, mockUtils);
+
+    Tasks.await(firebaseInstallations.getId());
+
+    PersistedFidEntry entryValue = persistedFid.readPersistedFidEntryValue();
+    assertThat(entryValue).hasFid(TEST_FID_1);
+
+    // Waiting for Task that registers FID on the FIS Servers
+    executor.awaitTermination(500, TimeUnit.MILLISECONDS);
+
+    // Validate that registration status is REGISTER_ERROR
+    PersistedFidEntry updatedFidEntry = persistedFid.readPersistedFidEntryValue();
+    assertThat(updatedFidEntry).hasFid(TEST_FID_1);
+    assertThat(updatedFidEntry).hasRegistrationStatus(RegistrationStatus.REGISTER_ERROR);
+  }
+
+  @Test
+  public void testGetId_expiredAuthTokenUncheckedException_statusUpdated() throws Exception {
+    // Update local storage with fid entry that has auth token expired.
+    persistedFid.insertOrUpdatePersistedFidEntry(EXPIRED_AUTH_TOKEN_ENTRY);
+    // Mocking unchecked exception on FIS generateAuthToken
+    when(mockClient.generateAuthToken(anyString(), anyString(), anyString(), anyString()))
+        .thenAnswer(
+            invocation -> {
+              throw new InterruptedException();
+            });
+    when(mockUtils.isAuthTokenExpired(any())).thenReturn(true);
+
+    FirebaseInstallations firebaseInstallations =
+        new FirebaseInstallations(executor, firebaseApp, mockClient, persistedFid, mockUtils);
+
+    assertWithMessage("getId Task failed")
+        .that(Tasks.await(firebaseInstallations.getId()))
+        .isNotEmpty();
+    PersistedFidEntry entryValue = persistedFid.readPersistedFidEntryValue();
+    assertThat(entryValue).hasFid(TEST_FID_1);
+
+    // Waiting for Task that generates auth token with the FIS Servers
+    executor.awaitTermination(500, TimeUnit.MILLISECONDS);
+
+    // Validate that registration status is REGISTER_ERROR
+    PersistedFidEntry updatedFidEntry = persistedFid.readPersistedFidEntryValue();
+    assertThat(updatedFidEntry).hasFid(TEST_FID_1);
+    assertThat(updatedFidEntry).hasRegistrationStatus(RegistrationStatus.REGISTER_ERROR);
+  }
+
+  @Test
+  public void testGetId_expiredAuthToken_refreshesAuthToken() throws Exception {
+    when(mockUtils.isAuthTokenExpired(any())).thenReturn(true);
```

Comment:
why does this happen first thing in this test but like 3rd in the previous test?

Reply:
No specific reason. Does the order matter?